tests/model/ArrayListTest.php

```diff
@@ -426,6 +426,25 @@ public function testMultiSort() {
 	}
 
 	/**
+	 * Check that we don't cause recursion errors with array_multisort() and customised ViewableData objects
+	 */
+	public function testSortWithRecursiveReferences() {
+		$items = ArrayList::create();
+
+		// Objects need the same sort value so that their properties are compared by array_multisort
+		$items->add(DataObject::create(array('Order' => 1)));
+		$items->add(DataObject::create(array('Order' => 1)));
+
+		// Customising the object adds recursive dependencies (between the object and its customised instance)
+		foreach ($items as $item) {
+			$item->customise(array());
```

Comment:
I’m not sure if it’s possible to set them up manually, but I didn’t want to anyway as we’re testing a specific scenario here - where customised `ViewableData` objects cause recursion errors. Setting them up manually is more leaning towards testing PHP’s behaviour with recursive/cyclic dependencies.

I don’t think this is likely to be broken unless we completely change how `ViewableData::customise()` works, in which case we’d probably _want_ this to break so that it highlights that we can (hopefully) remove the extra logic I added to `ArrayList::sort()`. If I set the dependencies up manually and `ViewableData::customise()` changes, then this test would continue passing (and be forgotten about) even if it’s completely redundant.